src/components/nabla/Pools/Backstop/WithdrawLiquidity/useWithdrawLiquidity.ts

```diff
@@ -2,91 +2,159 @@ import { yupResolver } from '@hookform/resolvers/yup';
+  const storageValues = storageService.getParsed<TransactionSettings>(storageKeys.POOL_SETTINGS);
+  return {
+    ...defaultValues,
+    amount: undefined,
+    address: null,
+    slippage: getValidSlippage(storageValues?.slippage),
+  };
+};
+
+function parseError(error: MessageCallErrorResult): string {
+  switch (error.type) {
+    case 'error':
+      return 'Cannot determine value of shares';
+    case 'panic':
+      return 'Cannot determine value of shares';
+    case 'reverted':
+      switch (error.description) {
+        case 'withdraw():MINIMUM_AMOUNT':
+        case 'withdrawExcessSwapLiquidity():MIN_AMOUNT':
+          return 'The returned amount of tokens is below your desired minimum amount.';
+        case 'sharesTargetWorth():POOLWORTH_NEGATIVE':
+          return 'It is not possible to withdraw from the backstop pool at the moment.';
+        case 'withdraw: INSUFFICIENT_BALANCE':
+        case 'withdrawExcessSwapLiquidity():BALANCE':
+          return "You don't have enough LP tokens to redeem.";
+        case 'SwapPool#backstopDrain():INSUFFICIENT_COVERAGE':
+          return 'The input amount is too large.';
+      }
+      return 'Cannot determine value of shares';
+  }
+}
 
 const storageSet = debounce(storageService.set, 1000);
-export const useWithdrawLiquidity = (pool: BackstopPool) => {
-  const { id: poolAddress, token, router } = pool;
-  const tokenAddress = token.id;
-  const swapPools = router?.swapPools;
-  const { indexerUrl } = useGetAppDataByTenant('nabla').data || {};
+
+export const useWithdrawLiquidity = (nabla: NablaInstance) => {
   const queryClient = useQueryClient();
   const toggle = useModalToggle();
-  const tokenModal = useBoolean();
 
-  const balanceQuery = useContractBalance({ contractAddress: tokenAddress, decimals: FixedU128Decimals });
-  const depositQuery = useContractBalance({ contractAddress: poolAddress, decimals: FixedU128Decimals });
-  const { refetch: balanceRefetch } = balanceQuery;
-  const { refetch: depositRefetch } = depositQuery;
+  const backstopPoolAddress = nabla.backstopPool.id;
+
+  const swapPools = nabla.swapPools;
 
   const form = useForm<WithdrawLiquidityValues>({
     resolver: yupResolver(schema),
-    defaultValues: {},
+    defaultValues: getInitialValues(),
   });
-  const { reset, getValues, handleSubmit } = form;
-  const amount =
-    Number(
-      useWatch({
-        control: form.control,
-        name: 'amount',
-        defaultValue: 0,
-      }),
-    ) || 0;
 
   const address = useWatch({
     control: form.control,
     name: 'address',
   });
 
-  const pools = useMemo(
-    () =>
-      [
-        {
-          id: '',
-          token: {
-            ...token,
-            id: '',
-          },
-        } as SwapPool,
-      ].concat(swapPools || []),
-    [swapPools, token],
+  const selectedPool = useMemo<NablaInstanceSwapPool | undefined>(
+    () => swapPools.find((t) => t.id === address)!,
```

Comment:
Is the "!" at the end necessary?

Reply:
It's generally best to avoid using TypeScript's non-null assertion as it can lead to runtime errors if our assumptions are incorrect. What do you think about adding a check for undefined here?

Reply:
This indeed does not make sense here. Removed.